### PR TITLE
Spec file cleanups

### PR DESCRIPTION
**Pull request by smithfarm**

### CODE REVIEW

**Comment by smithfarm** [comment]
@tserong Please review? And do you know if the directories need to be excluded, too?

**Comment by smithfarm** [comment]
Test-building now in https://build.suse.de/package/show/Devel:Storage:3.0:Staging:Test/ceph

**Comment by smithfarm** [comment]
```
[ 1900s] error: Installed (but unpackaged) file(s) found:
[ 1900s]    /usr/lib/debug/.build-id/1b/528478294bd2ebb96e6ca7a69f8a52e8c57200
[ 1900s]    /usr/lib/debug/.build-id/1b/528478294bd2ebb96e6ca7a69f8a52e8c57200.debug
[ 1900s]    /usr/lib/debug/.build-id/32/7066c70c569998480b2c49c1bff66d2f510b67
[ 1900s]    /usr/lib/debug/.build-id/32/7066c70c569998480b2c49c1bff66d2f510b67.debug
[ 1900s]    /usr/lib/debug/.build-id/51/43ec35bb95760aad5986aa25a7ee30b799a5bf
[ 1900s]    /usr/lib/debug/.build-id/51/43ec35bb95760aad5986aa25a7ee30b799a5bf.debug
[ 1900s]    /usr/lib/debug/.build-id/7f/dc95bbaa2de575802c062d48b5ca496bd824a1
[ 1900s]    /usr/lib/debug/.build-id/7f/dc95bbaa2de575802c062d48b5ca496bd824a1.debug
[ 1900s]    /usr/lib/debug/.build-id/a2/4e76bd1471dd683ac1b507187da2a97e2e5925
[ 1900s]    /usr/lib/debug/.build-id/a2/4e76bd1471dd683ac1b507187da2a97e2e5925.debug
[ 1900s]    /usr/lib/debug/.build-id/ac/0632f4b8ef257ff3f0a5045fbc03a7658edba0
[ 1900s]    /usr/lib/debug/.build-id/ac/0632f4b8ef257ff3f0a5045fbc03a7658edba0.debug
[ 1900s]    /usr/lib/debug/.build-id/b9/2e1635beacaa249993b0c340f60ade1d2ec65f
[ 1900s]    /usr/lib/debug/.build-id/b9/2e1635beacaa249993b0c340f60ade1d2ec65f.debug
[ 1900s]    /usr/lib/debug/.build-id/b9/657a434ebff3b75293c94806da2d86b69ddb5a
[ 1900s]    /usr/lib/debug/.build-id/b9/657a434ebff3b75293c94806da2d86b69ddb5a.debug
[ 1900s]    /usr/lib/debug/.build-id/c8/a4b169a3bf32870b1496c1f6475120b9b67f09
[ 1900s]    /usr/lib/debug/.build-id/c8/a4b169a3bf32870b1496c1f6475120b9b67f09.debug
[ 1900s]    /usr/lib/debug/.build-id/d4/3451c6ec1a970b51050b2029b8db67712fe6dd
[ 1900s]    /usr/lib/debug/.build-id/d4/3451c6ec1a970b51050b2029b8db67712fe6dd.debug
[ 1900s]    /usr/lib/debug/.build-id/e4/4ff7839daef361dffd5cc3087e2d9a5257fc8d
[ 1900s]    /usr/lib/debug/.build-id/e4/4ff7839daef361dffd5cc3087e2d9a5257fc8d.debug
[ 1900s]    /usr/lib/debug/.build-id/eb/af02af13fe5571cc5df1579bcc5b38338484a7
[ 1900s]    /usr/lib/debug/.build-id/eb/af02af13fe5571cc5df1579bcc5b38338484a7.debug
[ 1900s]    /usr/lib/debug/.build-id/ec/9972e35afc89ff197a025a2ab7127b75f44c0a
[ 1900s]    /usr/lib/debug/.build-id/ec/9972e35afc89ff197a025a2ab7127b75f44c0a.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/compressor/libceph_example.so.0.0.0.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/compressor/libceph_snappy.so.2.0.0.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/compressor/libceph_zlib.so.2.0.0.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_isa.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_jerasure.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_jerasure_generic.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_jerasure_sse3.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_jerasure_sse4.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_lrc.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_shec.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_shec_generic.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_shec_sse3.so.debug
[ 1900s]    /usr/lib/debug/usr/lib64/ceph/erasure-code/libec_shec_sse4.so.debug
```

**Comment by smithfarm** [comment]
Obviously, using `%exclude` on shared objects doesn't exclude them from debuginfo.

Now I'm trying a different tactic: I added a new `ceph-plugins-devel` package and moved the stuff into it from `ceph-base`.

**Comment by smithfarm** [comment]
Test-building now in https://build.suse.de/package/show/Devel:Storage:3.0:Staging:Test/ceph

**Comment by smithfarm** [comment]
@ddiss @kmroz @tserong

**Comment by smithfarm** [comment]
The purpose of this PR is not not clutter up ceph-base with files that we don't support and are not intended for end-user consumption.

File: ceph.spec.in [hunk 1]
```diff
@@ -379,6 +379,15 @@ developed as part of the Ceph distributed storage system. This is a
 shared library allowing applications to access the distributed object
 store using a simple file-like interface.
 
+%package -n ceph-plugins-devel
```
Comment (tserong) [inline ceph.spec.in]: Minor nit: there should be a line of hash symbols above this section, as with other %package sections

File: ceph.spec.in [hunk 2]
```diff
@@ -1238,6 +1238,23 @@ fi
 %postun -n librados2
 /sbin/ldconfig
 
+%files -n ceph-plugins-devel
```
Comment (tserong) [inline ceph.spec.in]: Minor nit: there should be a line of hash symbols above this section, as with other %files sections

**Comment by tserong** [comment]
Isn't the %{_libdir}/ceph/erasure-code stuff necessary if you're using EC pools? By which I mean, shouldn't that remain in base?

File: ceph.spec.in (hunk 1 above)
Comment (ddiss) [inline ceph.spec.in]: Another minor nit: you could drop the "-n ceph-" here - without "-n" it'll automatically pickup the "ceph-" prefix.

**Comment by smithfarm** [comment]
@tserong AFAIK the plugins (i.e. the so files in `/usr/lib/ceph`) are used by some of the make check tests and are otherwise not used at all. The debian packaging does not include any of the files this PR is moving to the new package.

**Comment by smithfarm** [comment]
@tserong I added a commit that drops all the `#######` divider lines. Doing the same upstream in #9757.

File: ceph.spec.in (hunk 1 above)
Comment (smithfarm) [inline ceph.spec.in]: done

**Comment by smithfarm** [comment]
Marking DNM until we reach a consensus on what branch to base SES4 on.

**Comment by kmroz** [comment]
@smithfarm - :+1: for moving these bits to a separate package.

**Comment by tserong** [comment]
> The debian packaging does not include any of the files this PR is moving to the new package.

I don't know what debian are doing, but if I remove /usr/lib64/ceph/erasure-code on my SES3 test VMs, the MONs won't start anymore:

````
2016-08-11 17:50:37.707330 7f4b9f6984c0 -1 load dlopen(/usr/lib64/ceph/erasure-code/libec_jerasure.so): /usr/lib64/ceph/erasure-code/libec_jerasure.so: cannot open shared object file: No such file or directory
````

**Comment by smithfarm** [comment]
@tserong Wow, good catch!

**Comment by smithfarm** [comment]
```
smithfarm@wilbur:~/src/ceph/SUSE/ceph/debian> grep -r 'erasure-code'
ceph-base.install:usr/lib/ceph/erasure-code/*
```

**Comment by smithfarm** [comment]
dlopen is a blight

**Comment by tserong** [comment]
So I'm thinking now we shouldn't move this stuff to a new package, the erasure code plugins because they're dlopen()'d by the mons, and the compressor stuff because I assume (but am not completely certain) that that might be used by bluestore, in which case in future when we *do* support bluestore, we'd just have to move them back (probably).

**Comment by smithfarm** [comment]
My understanding was that SES3/4 will never support bluestore - that it's a SES5 deliverable. But I agree I need to find out more about the compressor plugins before depriving customers of them.

**Comment by smithfarm** [comment]
Dropped the commit that was adding the "ceph-plugins-devel" package.

Now this is more-or-less a backport of https://github.com/ceph/ceph/pull/9757

**Comment by smithfarm** [comment]
@l-mb This builds. OK to merge to ses3 branch? We also discussed the idea of starting a ses4 branch - what do you think of that? (I tend to think we should develop ceph-for-SES4 in the ses3 branch for as long as possible.)

**Comment by smithfarm** [comment]
Test build (green) is here: https://build.suse.de/package/show/Devel:Storage:3.0:Staging:Test/ceph

**Pull request merged**